src/integration-test/java/de/danielbechler/diff/CategoriesTestIT.groovy

```diff
@@ -0,0 +1,133 @@
+package de.danielbechler.diff
+
+import de.danielbechler.diff.introspection.ObjectDiffProperty
+import de.danielbechler.diff.node.DiffNode
+import de.danielbechler.diff.node.Visit
+import de.danielbechler.diff.path.NodePath
+import spock.lang.Specification
+
+class CategoriesTestIT extends Specification{
+
+	def obj1 = new MyObject("aaa","aaa", "aaa")
+	def obj2 =  new MyObject("bbb","bbb", "bbb")
+	def differ = ObjectDifferBuilder.startBuilding()
+			.categories()
+			.ofNode(NodePath.with("firstString")).toBe("cat1")
+			.ofNode(NodePath.with("secondString")).toBe("cat1")
+			.ofNode(NodePath.with("thirdString")).toBe("cat1")
+			.and()
+			.build()
+
+	def node = differ.compare(obj1,obj2)
+
+	def categoriesMapVisitor = new DiffNode.Visitor() {
+
+		Map<String, Set<String>> mapCategories = new HashMap<>();
+
+		@Override
+		void node(DiffNode node, Visit visit) {
+
+			mapCategories.put(node.getPropertyName(), node.getCategories())
+		}
+	}
+
+	def categoriesAdderVisitor = new DiffNode.Visitor() {
+
+		@Override
+		void node(DiffNode node, Visit visit) {
+
+			node.addCategories(Arrays.asList("addedWhileVisiting"))
+		}
+	}
+
+	def "should return all categories"(){
+		given:
+			node.visitChildren(categoriesMapVisitor)
+	   expect :
+			categoriesMapVisitor.mapCategories.get("firstString") == ["cat1"] as Set
+			categoriesMapVisitor.mapCategories.get("secondString") == ["cat1"] as Set
+			categoriesMapVisitor.mapCategories.get("thirdString") == ["cat1", "catAnnotation"] as Set
+	}
```

Comment:
How about doing this instead of using the mapVisitor?

```groovy
node.getChild("firstString").getCategories() == ["cat1"] as Set
...
```

Or was there a specific reason for the visitor approach?